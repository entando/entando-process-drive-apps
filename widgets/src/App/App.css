.app-list {
  width: 250px;
  padding: 0 10px;
}

.app-home {
  text-align: center;
  padding: 60px;
}

.app-container {
  padding: 20px;
  background-color: #f3f3f3;
}
