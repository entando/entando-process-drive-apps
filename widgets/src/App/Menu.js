import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Typography from '@material-ui/core/Typography';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import Drawer from '@material-ui/core/Drawer';
import Divider from '@material-ui/core/Divider';
import IconButton from '@material-ui/core/IconButton';
import {
  Attachment as AttachmentIcon,
  Home as HomeIcon,
  ViewQuilt as ViewQuiltIcon,
  List as ListIcon,
  BarChart as BarChartIcon,
  Settings as SettingsIcon,
  Dashboard as DashboardIcon,
  Description as DescriptionIcon,
  DoneAll as DoneAllIcon,
  Comment as CommentsIcon,
  CloudQueue as CloudQueueIcon,
  DeveloperBoard as DeveloperBoardIcon,
} from '@material-ui/icons';
import { withStyles } from '@material-ui/core/styles';

const styles = {
  item: {
    position: 'relative',
  },
  listItem: {
    paddingLeft: 10,
    paddingRight: 10,
  },
  iconButton: {
    position: 'absolute',
    right: 5,
    top: 12,
  },
};

const menuItems = [
  {
    linkTo: '/Connections',
    linkText: 'Connections',
    Icon: <CloudQueueIcon />,
  },
  {
    linkTo: '/TaskList',
    linkText: 'Task list',
    Icon: <ListIcon />,
    configLinkTo: '/TaskListConfig',
  },
  {
    linkTo: '/TaskDetails',
    linkText: 'Task details',
    Icon: <DescriptionIcon />,
    configLinkTo: '/TaskDetailsConfig',
  },
  {
    linkTo: '/TaskCompletionForm',
    linkText: 'Task completion form',
    Icon: <DoneAllIcon />,
    configLinkTo: '/TaskCompletionFormConfig',
  },
  {
    linkTo: '/TaskComments',
    linkText: 'Task comments',
    Icon: <CommentsIcon />,
    configLinkTo: '/TaskCommentsConfig',
  },
  {
    linkTo: '/SummaryCard',
    linkText: 'Summary card',
    Icon: <DashboardIcon />,
    configLinkTo: '/SummaryCardConfig',
  },
  {
    linkTo: '/ProcessForm',
    linkText: 'Process form',
    Icon: <DescriptionIcon />,
    configLinkTo: '/ProcessFormConfig',
  },
  {
    linkTo: '/ProcessDefinition',
    linkText: 'Process Definition',
    Icon: <DeveloperBoardIcon />,
    configLinkTo: 'ProcessDefinitionConfig',
  },
  {
    linkTo: '/OvertimeGraph',
    linkText: 'Overtime graph',
    Icon: <BarChartIcon />,
    configLinkTo: '/OvertimeGraphConfig',
  },
  {
    linkTo: '/Attachments',
    linkText: 'Attachments',
    Icon: <AttachmentIcon />,
    configLinkTo: '/AttachmentsConfig',
  },
  {
    linkTo: '/ProcessList',
    linkText: 'Process List',
    Icon: <ListIcon />,
    configLinkTo: '/ProcessListConfig',
  },
];

const Menu = ({ open, setOpen, classes }) => (
  <Drawer open={open} onClose={() => setOpen(false)}>
    <div className="app-list">
      <List>
        <ListItem button component={Link} to="/" onClick={() => setOpen(false)}>
          <ListItemIcon>
            <HomeIcon fontSize="small" />
          </ListItemIcon>
          <ListItemText primary="Home" />
        </ListItem>

        <ListItem button component={Link} to="/task-details-page/" onClick={() => setOpen(false)}>
          <ListItemIcon>
            <ViewQuiltIcon fontSize="small" />
          </ListItemIcon>
          <ListItemText primary="Task Details page" />
        </ListItem>

        <ListItem button component={Link} to="/smart-inbox-page/" onClick={() => setOpen(false)}>
          <ListItemIcon>
            <ViewQuiltIcon fontSize="small" />
          </ListItemIcon>
          <ListItemText primary="Smart Inbox page" />
        </ListItem>

        <Typography variant="overline">Widgets</Typography>
        <Divider />
        {menuItems.map(({ linkTo, Icon, linkText, configLinkTo }) => (
          <div className={classes.item} key={linkTo}>
            <ListItem
              className={classes.listItem}
              button
              component={Link}
              to={linkTo}
              onClick={() => setOpen(false)}
            >
              {Icon}
              <ListItemText style={{ marginLeft: 5 }} primary={linkText} />
            </ListItem>
            {configLinkTo && (
              <IconButton
                className={classes.iconButton}
                size="small"
                component={Link}
                to={configLinkTo}
                onClick={() => setOpen(false)}
              >
                <SettingsIcon fontSize="small" />
              </IconButton>
            )}
          </div>
        ))}
      </List>
    </div>
  </Drawer>
);

Menu.propTypes = {
  classes: PropTypes.shape({
    item: PropTypes.string,
    listItem: PropTypes.string,
    iconButton: PropTypes.string,
  }),
  open: PropTypes.bool,
  setOpen: PropTypes.func,
};

Menu.defaultProps = {
  classes: {},
  open: false,
  setOpen: () => {},
};

export default withStyles(styles)(Menu);
