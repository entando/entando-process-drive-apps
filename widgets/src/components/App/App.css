.App {
  text-align: center;
}

.app-list {
  width: 250px;
  padding: 0 10px;
}

.app-home {
  padding: 60px;
}

.app-container {
  padding: 20px;
}
