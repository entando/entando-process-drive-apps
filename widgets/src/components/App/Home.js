import React from 'react';
import { Typography } from '@material-ui/core';

const Home = () => (
  <div className="app-home">
    <Typography variant="h1" style={{ fontSize: '1.5rem' }}>
      Entando Process Automation Manager
    </Typography>
    <Typography variant="body1">
      This is a demo page of all available Entando PAM widgets
    </Typography>
  </div>
);

export default Home;
