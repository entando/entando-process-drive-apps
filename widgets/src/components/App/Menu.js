import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import {
  Typography,
  List,
  ListItem,
  ListItemText,
  ListItemIcon,
  Drawer,
  Divider,
  IconButton,
} from '@material-ui/core';
import {
  Home as HomeIcon,
  List as ListIcon,
<<<<<<< HEAD
  Settings as SettingsIcon,
  Dashboard as DashboardIcon,
  Description as DescriptionIcon,
  DoneAll as DoneAllIcon,
  Comment as CommentsIcon,
=======
  BarChart as BarChartIcon,
  Settings as SettingsIcon,
>>>>>>> PDA-71 Fix styles
} from '@material-ui/icons';
import { withStyles } from '@material-ui/core/styles';

const styles = {
  item: {
    position: 'relative',
  },
  iconButton: {
    position: 'absolute',
    right: 5,
    top: 11,
  },
};

const menuItems = [
  {
    linkTo: '/TaskList',
    linkText: 'Task list',
    Icon: <ListIcon />,
    configLinkTo: '/TaskListConfig',
  },
  {
    linkTo: '/TaskDetails',
    linkText: 'Task details',
    Icon: <DescriptionIcon />,
    configLinkTo: '/TaskDetailsConfig',
  },
  {
    linkTo: '/TaskCompletionForm',
    linkText: 'Task completion form',
    Icon: <DoneAllIcon />,
    configLinkTo: '/TaskCompletionFormConfig',
  },
  {
    linkTo: '/TaskComments',
    linkText: 'Task comments',
    Icon: <CommentsIcon />,
    configLinkTo: '/TaskCommentsConfig',
  },
  {
    linkTo: '/SummaryCard',
    linkText: 'Summary card',
    Icon: <DashboardIcon />,
    configLinkTo: '/SummaryCardConfig',
  },
  {
    linkTo: '/ProcessForm',
    linkText: 'Process form',
    Icon: <DescriptionIcon />,
    configLinkTo: '/ProcessFormConfig',
  },
  {
    linkTo: '/OvertimeGraph',
    linkText: 'Overtime graph',
    Icon: <BarChartIcon />,
    configLinkTo: '/OvertimeGraphConfig',
  },
];

const Menu = ({ open, setOpen, classes }) => (
  <Drawer open={open} onClose={() => setOpen(false)}>
    <div className="app-list">
      <List>
        <ListItem button component={Link} to="/" onClick={() => setOpen(false)}>
          <ListItemIcon>
            <HomeIcon />
          </ListItemIcon>
          <ListItemText primary="Home" />
        </ListItem>

        <Typography variant="overline">Widgets</Typography>
        <Divider />
        {menuItems.map(({ linkTo, Icon, linkText, configLinkTo }) => (
          <div className={classes.item} key={linkTo}>
            <ListItem button component={Link} to={linkTo} onClick={() => setOpen(false)}>
              {Icon}
              <ListItemText primary={linkText} />
            </ListItem>
            <IconButton
              className={classes.iconButton}
              size="small"
              component={Link}
              to={configLinkTo}
              onClick={() => setOpen(false)}
            >
              <SettingsIcon fontSize="small" />
            </IconButton>
          </div>
        ))}
      </List>
    </div>
  </Drawer>
);

Menu.propTypes = {
  classes: PropTypes.shape({
    item: PropTypes.string,
    iconButton: PropTypes.string,
  }),
  open: PropTypes.bool.isRequired,
  setOpen: PropTypes.func.isRequired,
};

Menu.defaultProps = {
  classes: {},
};

export default withStyles(styles)(Menu);
