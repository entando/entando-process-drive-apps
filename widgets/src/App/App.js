import React, {useState} from 'react';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import AppBar from '@material-ui/core/AppBar';
import Checkbox from '@material-ui/core/Checkbox';
import Container from '@material-ui/core/Container';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import { Menu as MenuIcon } from '@material-ui/icons';
import { makeStyles } from '@material-ui/core/styles';

import { authenticate } from '../api/app-builder/pages';
import WIDGETS_CONFIG from '../mocks/app-builder/widgets';
import SETTINGS from '../mocks/app-builder/pages';
import Menu from './Menu';
import WidgetConfig from '../components/common/WidgetConfig';

import './App.css';

// pages
import TaskDetailsPage from '../pages/task-details/';
import SmartInboxPage from '../pages/smart-inbox/';

// widgets
import Home from './Home';
import ConnectionsContainer from '../components/Connections/ConnectionsContainer';

import TaskListContainer from '../components/TaskList/TaskListContainer';
import TaskListConfig from '../components/TaskList/TaskListConfig';
import OvertimeGraphContainer from '../components/OvertimeGraph/OvertimeGraphContainer';
import OvertimeGraphConfig from '../components/OvertimeGraph/OvertimeGraphConfig';

import SummaryCardContainer from '../components/SummaryCard/SummaryCardContainer';
import SummaryCardConfig from '../components/SummaryCard/SummaryCardConfig';

import TaskDetailsContainer from '../components/TaskDetails/TaskDetailsContainer';
import TaskDetailsConfig from '../components/TaskDetails/TaskDetailsConfig';

import TaskCompletionFormContainer from '../components/TaskCompletionForm/TaskCompletionFormContainer';
import TaskCompletionFormConfig from '../components/TaskCompletionForm/TaskCompletionFormConfig';

import TaskCommentsContainer from '../components/TaskComments/TaskCommentsContainer';
import TaskCommentsConfig from '../components/TaskComments/TaskCommentsConfig';

import ProcessFormContainer from '../components/ProcessForm/ProcessFormContainer';
import ProcessFormConfig from '../components/ProcessForm/ProcessFormConfig';

import AttachmentsContainer from '../components/Attachments/AttachmentsContainer';
import AttachmentsConfig from '../components/Attachments/AttachmentsConfig';

import ProcessDefinitionContainer from '../components/ProcessDefinition/ProcessDefinitionContainer';
import ProcessDefinitionConfig from '../components/ProcessDefinition/ProcessDefinitionConfig';

import ProcessListContainer from '../components/ProcessList/ProcessListContainer';
import ProcessListConfig from '../components/ProcessList/ProcessListConfig';

const useStyles = makeStyles(theme => ({
  appBar: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
    fontSize: '1.5rem',
  },
  userAuth: {
    marginLeft: '20px',
  },
}));

function App() {
  const [open, setOpen] = useState(false);
  const [lazyLoading, setLazyLoading] = useState(true);
  const [skeletonLoading, setSkeletonLoading] = useState(false);

  // TODO: Remove when token is managed by wrapper
  React.useEffect(() => {
    async function fetchToken() {
      // temporary fetch token for ease of development
      console.log('Fetching authentication token for ease of development! DEV ONLY'); // eslint-disable-line no-console

      const authentication = await authenticate();
      if (authentication && authentication.access_token) {
        console.log('New authentication token is:', authentication.access_token); // eslint-disable-line no-console
        localStorage.setItem('token', authentication.access_token);
      }
    }
    fetchToken();
  }, []);

  const classes = useStyles();

  return (
    <div className="App">
      <Router>
        <Routes>
          <Route path="/task-details-page/" element={<TaskDetailsPage />}/>
          <Route path="/smart-inbox-page/" element={<SmartInboxPage />}/>

          <Route path="*" element={
              <>
            <AppBar position="static" className={classes.appBar}>
              <Toolbar>
                <IconButton
                  edge="start"
                  color="inherit"
                  aria-label="menu"
                  onClick={() => setOpen(true)}
                  className={classes.menuButton}
                >
                  <MenuIcon />
                </IconButton>
                <Typography variant="h2" className={classes.title}>
                  Entando - PAM Plugin
                </Typography>
                <FormControlLabel
                  control={
                    <Checkbox
                      checked={skeletonLoading}
                      onChange={() => setSkeletonLoading(!skeletonLoading)}
                    />
                  }
                  label="Show skeleton"
                  labelPlacement="start"
                />
                <FormControlLabel
                  control={
                    <Checkbox checked={lazyLoading} onChange={() => setLazyLoading(!lazyLoading)} />
                  }
                  label="Lazy Loading"
                  labelPlacement="start"
                />
                <user-auth
                  kc-auth-url="http://test-keycloak.51.91.30.184.nip.io/auth"
                  kc-realm="entando"
                  kc-client-id="eti-dig-ex"
                  base-url=""
                  class={classes.userAuth}
                />
              </Toolbar>
            </AppBar>

            <Menu open={open} setOpen={setOpen} />

            <Container className="app-container" maxWidth={false}>
                <Routes>
                  <Route exact path="/" element={<Home/>} />

                  <Route exact path="/Connections" element={<ConnectionsContainer/>} />

                  <Route
                    path="/TaskList/"
                    element={
                      <TaskListContainer
                        lazyLoading={lazyLoading}
                        pageCode={WIDGETS_CONFIG.TASK_LIST.pageCode}
                        frameId={WIDGETS_CONFIG.TASK_LIST.frameId}
                      />
                    }
                  />
                  <Route
                    path="/TaskListConfig"
                    element={
                      <WidgetConfig
                        pageCode={WIDGETS_CONFIG.TASK_LIST.pageCode}
                        frameId={WIDGETS_CONFIG.TASK_LIST.frameId}
                        // hardcoded config can be passed here as well
                      >
                        <TaskListConfig />
                      </WidgetConfig>
                    }
                  />

                  <Route
                    path="/TaskDetails/"
                    element={
                      <TaskDetailsContainer
                        taskId={WIDGETS_CONFIG.TASK_DETAILS.taskId}
                        pageCode={WIDGETS_CONFIG.TASK_DETAILS.pageCode}
                        frameId={WIDGETS_CONFIG.TASK_DETAILS.frameId}
                        widgetCode={WIDGETS_CONFIG.TASK_DETAILS.widgetCode}
                        taskPos={0}
                        lastPage={0}
                        groups={SETTINGS.TASK_LIST.payload.config.groups}
                      />
                    }
                  />
                  <Route
                    path="/TaskDetailsConfig"
                    element={
                      <WidgetConfig
                        pageCode={WIDGETS_CONFIG.TASK_DETAILS.pageCode}
                        frameId={WIDGETS_CONFIG.TASK_DETAILS.frameId}
                      >
                        <TaskDetailsConfig />
                      </WidgetConfig>
                    }
                  />

                  <Route
                    path="/TaskCompletionForm/"
                    element={
                      <TaskCompletionFormContainer
                        taskId={WIDGETS_CONFIG.COMPLETION_FORM.taskId}
                        pageCode={WIDGETS_CONFIG.COMPLETION_FORM.pageCode}
                        frameId={WIDGETS_CONFIG.COMPLETION_FORM.frameId}
                        widgetCode={WIDGETS_CONFIG.COMPLETION_FORM.widgetCode}
                      />
                    }
                  />
                  <Route
                    path="/TaskCompletionFormConfig"
                    element={
                      <WidgetConfig
                        pageCode={WIDGETS_CONFIG.COMPLETION_FORM.pageCode}
                        frameId={WIDGETS_CONFIG.COMPLETION_FORM.frameId}
                      >
                        <TaskCompletionFormConfig />
                      </WidgetConfig>
                    }
                  />

                  <Route
                    path="/TaskComments/"
                    element={
                      <TaskCommentsContainer
                        taskId={WIDGETS_CONFIG.TASK_COMMENTS.taskId}
                        pageCode={WIDGETS_CONFIG.TASK_COMMENTS.pageCode}
                        frameId={WIDGETS_CONFIG.TASK_COMMENTS.frameId}
                        widgetCode={WIDGETS_CONFIG.TASK_COMMENTS.widgetCode}
                      />
                    }
                  />
                  <Route
                    path="/TaskCommentsConfig"
                    element = {
                      <WidgetConfig
                        pageCode={WIDGETS_CONFIG.TASK_COMMENTS.pageCode}
                        frameId={WIDGETS_CONFIG.TASK_COMMENTS.frameId}
                      >
                        <TaskCommentsConfig />
                      </WidgetConfig>
                    }
                  />

                  <Route
                    path="/SummaryCard/"
                    element={
                      <SummaryCardContainer
                        pageCode={WIDGETS_CONFIG.SUMMARY_CARD.pageCode}
                        frameId={WIDGETS_CONFIG.SUMMARY_CARD.frameId}
                        widgetCode={WIDGETS_CONFIG.SUMMARY_CARD.widgetCode}
                      />
                    }
                  />
                  <Route
                    path="/SummaryCardConfig/"
                    element={
                      <WidgetConfig
                        pageCode={WIDGETS_CONFIG.SUMMARY_CARD.pageCode}
                        frameId={WIDGETS_CONFIG.SUMMARY_CARD.frameId}
                      >
                        <SummaryCardConfig />
                      </WidgetConfig>
                    }
                  />

                  <Route
                    path="/ProcessForm"
                    element={
                      <ProcessFormContainer
                        pageCode={WIDGETS_CONFIG.PROCESS_FORM.pageCode}
                        frameId={WIDGETS_CONFIG.PROCESS_FORM.frameId}
                        widgetCode={WIDGETS_CONFIG.PROCESS_FORM.widgetCode}
                      />
                    }
                  />
                  <Route
                    path="/ProcessFormConfig"
                    element={
                      <WidgetConfig
                        pageCode={WIDGETS_CONFIG.PROCESS_FORM.pageCode}
                        frameId={WIDGETS_CONFIG.PROCESS_FORM.frameId}
                      >
                        <ProcessFormConfig />
                      </WidgetConfig>
                    }
                  />

                  <Route
                    path="/OvertimeGraph"
                    element={
                      <OvertimeGraphContainer
                        pageCode={WIDGETS_CONFIG.OVERTIME_GRAPH.pageCode}
                        frameId={WIDGETS_CONFIG.OVERTIME_GRAPH.frameId}
                        widgetCode={WIDGETS_CONFIG.OVERTIME_GRAPH.widgetCode}
                      />
                    }
                  />
                  <Route
                    path="/OvertimeGraphConfig"
                    element={
                      <WidgetConfig
                        pageCode={WIDGETS_CONFIG.OVERTIME_GRAPH.pageCode}
                        frameId={WIDGETS_CONFIG.OVERTIME_GRAPH.frameId}
                      >
                        <OvertimeGraphConfig />
                      </WidgetConfig>
                    }
                  />

                  <Route
                    path="/Attachments"
                    element={
                      <AttachmentsContainer
                        taskId={WIDGETS_CONFIG.ATTACHMENTS.taskId}
                        pageCode={WIDGETS_CONFIG.ATTACHMENTS.pageCode}
                        frameId={WIDGETS_CONFIG.ATTACHMENTS.frameId}
                        widgetCode={WIDGETS_CONFIG.ATTACHMENTS.widgetCode}
                      />
                    }
                  />
                  <Route
                    path="/AttachmentsConfig"
                    element={
                      <WidgetConfig
                        pageCode={WIDGETS_CONFIG.ATTACHMENTS.pageCode}
                        frameId={WIDGETS_CONFIG.ATTACHMENTS.frameId}
                      >
                        <AttachmentsConfig />
                      </WidgetConfig>
                    }
                  />

                  <Route
                    path="/ProcessDefinition"
                    element={
                      <ProcessDefinitionContainer
                        taskId={WIDGETS_CONFIG.PROCESS_DEFINITION.taskId}
                        pageCode={WIDGETS_CONFIG.PROCESS_DEFINITION.pageCode}
                        frameId={WIDGETS_CONFIG.PROCESS_DEFINITION.frameId}
                        widgetCode={WIDGETS_CONFIG.PROCESS_DEFINITION.widgetCode}
                      />
                    }
                  />
                  <Route
                    path="/ProcessDefinitionConfig"
                    element={
                      <WidgetConfig
                        pageCode={WIDGETS_CONFIG.PROCESS_DEFINITION.pageCode}
                        frameId={WIDGETS_CONFIG.PROCESS_DEFINITION.frameId}
                      >
                        <ProcessDefinitionConfig />
                      </WidgetConfig>
                    }
                  />

                  <Route
                    path="/ProcessList"
                    element={
                      <ProcessListContainer
                        taskId={WIDGETS_CONFIG.PROCESS_LIST.taskId}
                        pageCode={WIDGETS_CONFIG.PROCESS_LIST.pageCode}
                        frameId={WIDGETS_CONFIG.PROCESS_LIST.frameId}
                        widgetCode={WIDGETS_CONFIG.PROCESS_LIST.widgetCode}
                      />
                    }
                  />
                  <Route
                    path="/ProcessListConfig"
                    element={
                      <WidgetConfig
                        pageCode={WIDGETS_CONFIG.PROCESS_LIST.pageCode}
                        frameId={WIDGETS_CONFIG.PROCESS_LIST.frameId}
                      >
                        <ProcessListConfig />
                      </WidgetConfig>
                    }
                  />
                </Routes>
              </Container>
            </>
          }>
          </Route>
        </Routes>
      </Router>
    </div>
  );
}

export default App;
