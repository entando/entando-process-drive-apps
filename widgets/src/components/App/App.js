import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import {
  AppBar,
  Checkbox,
  Container,
  FormControlLabel,
  Toolbar,
  Typography,
  IconButton,
} from '@material-ui/core';
import { Menu as MenuIcon } from '@material-ui/icons';
import { makeStyles } from '@material-ui/core/styles';

import 'components/App/App.css';
import Menu from 'components/App/Menu';

// widgets
import Home from 'components/App/Home';

import TaskListContainer from 'components/TaskList/TaskListContainer';
import TaskListConfig from 'components/TaskList/TaskListConfig';

import TaskDetailsContainer from 'components/TaskDetails/TaskDetailsContainer';

const useStyles = makeStyles(theme => ({
  appBar: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
  },
}));

function App() {
  const [open, setOpen] = React.useState(false);
  const [lazyLoading, setLazyLoading] = React.useState(false);
  const classes = useStyles();

  return (
    <div className="App">
      <Router>
        <AppBar position="static" className={classes.appBar}>
          <Toolbar>
            <IconButton
              edge="start"
              color="inherit"
              aria-label="menu"
              onClick={() => setOpen(true)}
              className={classes.menuButton}
            >
              <MenuIcon />
            </IconButton>
            <Typography variant="h6" className={classes.title}>
              Entando - PAM Plugin
            </Typography>
            <FormControlLabel
              control={
                <Checkbox checked={lazyLoading} onChange={() => setLazyLoading(!lazyLoading)} />
              }
              label="Lazy Loading"
              labelPlacement="start"
            />
          </Toolbar>
        </AppBar>

        <Menu open={open} setOpen={setOpen} />

        <Container className="app-container">
          <Route path="/" exact component={Home} />
          <Route path="/TaskList/" render={() => <TaskListContainer lazyLoading={lazyLoading} />} />
          <Route
            path="/TaskListConfig"
            render={() => <TaskListConfig pageCode="0" framePos="0" widgetCode="pda_task_list" />}
          />
          <Route
            path="/TaskDetails/"
            render={() => <TaskDetailsContainer taskId="290" pageCode="0" framePos="0" />}
          />
        </Container>
      </Router>
    </div>
  );
}

export default App;
