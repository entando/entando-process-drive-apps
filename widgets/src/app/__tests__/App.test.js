import { render, wait } from '@testing-library/react';
import React from 'react';

import 'mocks/i18nMock';

import App from 'app/App';

describe('<App />', () => {
  it('renders snapshot correctly', async () => {
    fetch.once(JSON.stringify({ access_token: 'abc' }));
    const { container } = render(<App />);

    await wait(() => expect(container).toMatchSnapshot());
  });
});
