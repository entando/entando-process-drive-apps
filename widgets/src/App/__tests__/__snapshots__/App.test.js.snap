// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders snapshot correctly 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-appBar-1 MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <h2
          class="MuiTypography-root makeStyles-title-3 MuiTypography-h2"
        >
          Entando - PAM Plugin
        </h2>
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-111 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-114"
                data-indeterminate="false"
                type="checkbox"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Show skeleton
          </span>
        </label>
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-111 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-112 Mui-checked MuiIconButton-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input-114"
                data-indeterminate="false"
                type="checkbox"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Lazy Loading
          </span>
        </label>
        <user-auth
          base-url=""
          class="makeStyles-userAuth-4"
          kc-auth-url="http://test-keycloak.51.91.30.184.nip.io/auth"
          kc-client-id="eti-dig-ex"
          kc-realm="entando"
        />
      </div>
    </header>
    <div
      class="MuiContainer-root app-container"
    >
      <div
        class="app-home"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1"
          style="font-size: 1.5rem;"
        >
          Entando Process Automation Manager
        </h1>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          This is a demo page of all available Entando PAM widgets
        </p>
      </div>
    </div>
  </div>
</div>
`;
